src/TinyHelpers.AspNetCore/Attributes/FileSizeAttribute.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Http;
+
+namespace TinyHelpers.AspNetCore.Attributes;
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
+public class FileSizeAttribute : ValidationAttribute
```

Comment:
Come si relaziona questo attributo con l'attributo già esistente [RequestSizeLimit](https://khalidabuhakmeh.com/increase-file-upload-limit-for-aspdotnet)?

Reply:
Intendi se il file supera la body size di default di 30 MB?

Reply:
È più una domanda generica, che forse avrei dovuto fare fin dall'inizio (per questo ti chiedo scusa): Con l'attributo RequestSizeLimit non si riesce già, di fatto, ad ottenere la validazione sulla lunghezza del file? O ci sono scenari che non sono coperti?

Reply:
Da come mi è parso di capire (non l'ho mai usato) l'attributo RequestSizeLimit è per la size del body della request.

Immagino uno scenario dove ho più di un file. Come faccio a determinare la taglia massima di ogni singolo file presente nella request?

Reply:
Ok, quindi questo attributo ha senso se ho un endpoint che prende in input più file? Però non un IFormCollection, proprio due parametri IFormFile. È questo che intendi? Sto cercando lo scenario di applicabilità, perdonami se ti faccio tutte queste domande 😀

Reply:
Figurati, è un piacere risponderti (non essendo C# il linguaggio che uso di default, ho solo da imparare)

Comunque si, immaginavo due o più parametri IFormFile :) 
(e non la collection di files in IFormCollection)

Reply:
Ok, te l'ho chiesto perché questo scenario è abbastanza raro, tipicamente un endpoint prende in input o un singolo file oppure una collezione, non ho mai avuto a che fare con endpoint che richiedono espressamente degli IFormFile diversi, ad essere sincero non sono sicuro che a livello di binding sia uno scenario supportato. Riesci a fare un test per verificarne la fattibilità?

Reply:
Non basterebbe uno scenario come questo ?

Immagina che ho questa classe

```
public class Candidate
  {
      [FileSize(4000)]
      public IFormFile Resume { get; set; }


      [FileSize(2000)]
      public IFormFile ApplicationLetter { get; set; }
  }
    
```

Nel controller ho questa azione 

```
public async Task<IActionResult> SaveFiles([FromForm] Candidate candidate)
{
    // do stuff
}

```

Reply:
Questo appunto a livello di dichiarazione, ma eseguendo un test dell'applicazione (va bene anche con Swagger) ottieni il risultato sperato, quindi puoi caricare e recuperare i due file e per ognuno viene correttamente verificato il rispetto della lunghezza massima?

Reply:
Yep 🙂

Nella classe ho questa configurazione 
```
[FileSize(3450)]
public IFormFile Resume { get; set; }


[FileSize(2790)] 
public IFormFile ApplicationLetter { get; set; }
```

Provando a fare l'upload di un file di 3.5K, entrambi si ribellano 🙂 .
(Se hai piacere [qui](https://watch.screencastify.com/v/agLawfD5n56uE0zg86eK) c'è il video dove, probabilmente, si capisce meglio cosa intendo).